client/gateway_client.go

```diff
@@ -0,0 +1,394 @@
+package client
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/conduktor/ctl/resource"
+	"github.com/conduktor/ctl/schema"
+	"github.com/conduktor/ctl/utils"
+	"github.com/go-resty/resty/v2"
+)
+
+type GatewayClient struct {
+	cdkGatewayUser     string
+	cdkGatewayPassword string
+	baseUrl            string
+	client             *resty.Client
+	kinds              schema.KindCatalog
+}
+
+type GatewayApiParameter struct {
+	BaseUrl            string
+	Debug              bool
+	CdkGatewayUser     string
+	CdkGatewayPassword string
+}
+
+func MakeGateway(apiParameter GatewayApiParameter) (*GatewayClient, error) {
+	restyClient := resty.New().SetDebug(apiParameter.Debug).SetHeader("X-CDK-CLIENT", "CLI/"+utils.GetConduktorVersion())
+
+	if apiParameter.BaseUrl == "" {
+		return nil, fmt.Errorf("Please set CDK_GATEWAY_BASE_URL")
+	}
+
+	if apiParameter.CdkGatewayUser == "" || apiParameter.CdkGatewayPassword == "" {
+		return nil, fmt.Errorf("CDK_GATEWAY_USER and CDK_GATEWAY_PASSWORD must be provided")
+	}
+
+	result := &GatewayClient{
+		cdkGatewayUser:     apiParameter.CdkGatewayUser,
+		cdkGatewayPassword: apiParameter.CdkGatewayPassword,
+		baseUrl:            apiParameter.BaseUrl,
+		client:             restyClient,
+		kinds:              nil,
+	}
+
+	result.client.SetTLSClientConfig(&tls.Config{InsecureSkipVerify: true})
+	result.client.SetDisableWarn(true)
```

Comment:
@fteychene Is it always needed?

Reply:
I imagine that if @achever1 kept it by default it's that there is something about those warn
I can test some setup to check if we could remove it